bee-message/src/output/cpt2/signature_locked_single.rs

```diff
@@ -0,0 +1,44 @@
+// Copyright 2021-2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{address::Address, output::OutputAmount, Error};
+
+use packable::Packable;
+
+/// Describes an extended output with optional features.
+#[derive(Clone, Debug, Eq, PartialEq, Ord, PartialOrd, Packable)]
+#[cfg_attr(feature = "serde1", derive(serde::Serialize, serde::Deserialize))]
+#[packable(unpack_error = Error)]
+#[cfg_attr(doc_cfg, doc(cfg(feature = "cpt2")))]
+pub struct SignatureLockedSingleOutput {
+    address: Address,
+    // Amount of IOTA tokens held by the output.
+    #[packable(unpack_error_with = Error::InvalidOutputAmount)]
+    amount: OutputAmount,
+}
+
+impl SignatureLockedSingleOutput {
+    /// The [`Output`](crate::output::Output) kind of an [`SignatureLockedSingleOutput`].
+    pub const KIND: u8 = 0;
```

Comment:
In addition to the question from @pvdrz: I'm guessing the `KIND`s of the old types can lead to some confusion, because they might overlap now with the new ones.

Should we maybe mark them as `#[deprecated()]` with a warning to not use them any more? We could then explicitly `#[allow(deprecated)]` them where there are safe to use.

Reply:
They don't overlap, we kept increasing so these are still available

Reply:
If that's the case for all types then I'm fine with how it currently is.

Reply:
maybe I am not getting it, but how can they overlap with what Alex said here:
"Given that all of the variants are tagged with KIND values that do not differ between enabled/disabled".?

Reply:
Could it be the case that they start to overlap in future if the protocol and TIPs keep on changing?

Reply:
I think we'll always make it so it's backward compatible in terms of KINDs